execution/src/posix/execution.cc

```diff
@@ -188,11 +189,38 @@ namespace leatherman { namespace execution {
                 }
                 pipe.buffer.resize(4096);
             }
+            if (input.first != -1) {
```

Comment:
This can be reworked to share implementation by having a flag in pipe to switch between read/write.